### PR TITLE
Use sessionTimeout for session timeout.

### PR DESCRIPTION
**Pull request by aquamatthias**
Use a timer for the storage connection.

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/MarathonModule.scala [hunk 1]
```diff
@@ -140,10 +141,9 @@ class MarathonModule(conf: MarathonConf, http: HttpConf, zk: ZooKeeperClient)
   @Singleton
   def provideStore(): PersistentStore = {
     def directZK(): PersistentStore = {
-      implicit val timer = com.twitter.util.Timer.Nil
       import com.twitter.util.TimeConversions._
-      val sessionTimeout = conf.zooKeeperSessionTimeout.get.map(_.millis).getOrElse(30.minutes)
-      val connector = NativeConnector(conf.zkHosts, None, sessionTimeout, timer)
+      val sessionTimeout = conf.zooKeeperSessionTimeout().millis
+      val connector = NativeConnector(conf.zkHosts, None, sessionTimeout, new JavaTimer(true))
```
Comment (gkleiman) [inline src/main/scala/mesosphere/marathon/MarathonModule.scala]: Please include the name of the parameter when creating the JavaTimer (isDaemon).

File: src/main/scala/mesosphere/marathon/ZookeeperConf.scala [hunk 2]
```diff
@@ -17,11 +17,11 @@ trait ZookeeperConf extends ScallopConf {
 
   lazy val zooKeeperTimeout = opt[Long]("zk_timeout",
```
Comment (gkleiman) [inline src/main/scala/mesosphere/marathon/ZookeeperConf.scala]: What's the difference between these two timeouts?

We should add more info to the descriptions.

Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/ZookeeperConf.scala]: We used the zktimeout also as session timeout, which is just wrong.

**Comment by gkleiman** [comment]
Brilliant!

**Pull request merged**